src/main/scala/org/lyranthe/fs2_mongodb/MongoClient.scala

```diff
@@ -5,16 +5,18 @@ import fs2._
 import cats.effect.Sync
 
 object Mongo {
+
   def fromUrl[F[_]](url: String)(implicit F: Sync[F]): Stream[F, MongoClient] = {
-    Stream.bracket(F.delay(MongoClients.create(url)))(Stream.emit(_), { client =>
-      F.delay(client.close())
-    })
+    createStream(MongoClients.create(url))
+  }
+
+  def fromSettings[F[_]](settings: MongoClientSettings)(implicit F: Sync[F]): Stream[F, MongoClient] = {
+    createStream(MongoClients.create(settings))
   }
 
-  def fromSettings[F[_]](settings: MongoClientSettings)(
-      implicit F: Sync[F]): Stream[F, MongoClient] = {
-    Stream.bracket(F.delay(MongoClients.create(settings)))(Stream.emit(_), { client =>
+  private def createStream[F[_]](mongoClient: => MongoClient)(implicit F: Sync[F]): Stream[F, MongoClient] = {
```

Comment:
What do you think about exposing a `Resource` instead of a stream?

Reply:
I think, yes, it's probably the more idiomatic thing to do at this point (as much as there are idioms with unreleased software).